packages/upload-api/test/provisions-storage.js

```diff
@@ -71,7 +76,65 @@ export class ProvisionsStorage {
     const exists = Object.values(this.provisions).find(
       (p) => p.provider === provider && p.customer === customer
     )
-    return { ok: exists ? { did: customer } : null }
+    return exists
+      ? { ok: { did: customer } }
+      : {
+          error: {
+            name: 'CustomerNotFound',
+            message: 'customer does not exist',
+          },
+        }
+  }
+
+  /**
+   *
+   * @param {Types.ProviderDID} provider
+   * @param {string} subscription
+   * @returns
+   */
+  async getSubscription(provider, subscription) {
+    const provision = Object.values(this.provisions).find(
+      (p) => p.customer === subscription && p.provider === provider
+    )
+    if (provision) {
+      return { ok: provision }
+    } else {
+      return {
+        error: {
+          name: 'SubscriptionNotFound',
+          message: `could not find ${subscription}`,
+        },
+      }
+    }
+  }
+
+  /**
+   *
+   * @param {Types.ProviderDID} provider
+   * @param {*} consumer
```

Comment:
Is this a string or something ? `*` is same as `any` which is not ideal. You could also put `unknown` if you mean some unknown value which I think will work here.

Reply:
whoop - autogenerated, should be a string, good catch!